src/compiler/types.ts

```diff
@@ -1861,6 +1861,53 @@ namespace ts {
         getJsxIntrinsicTagNames(): Symbol[];
         isOptionalParameter(node: ParameterDeclaration): boolean;
 
+        /**
+         * Two types are identical iff they are references to the exact same fully qualified, resolved type (usually an alias to ===)
+         */
+        isIdenticalTo(a: Type, b: Type): boolean;
+        /**
+         * A type is a subtype of another if the collection of its members is contained within the collection of the other's members
+         */
+        isSubtypeOf(a: Type, b: Type): boolean;
+        /**
+         * Same as subtype, but all types are assignable to `any` and `any` is assignable to all types

```

Comment:
I agree, just pointing out that if people want the specifics, there is a formal description they can view.